morphir/src/org/finos/morphir/runtime/RTValue.scala

```diff
@@ -642,24 +642,24 @@ object RTValue {
     }
   }
   case class Applied(
-      body: RuntimeValue,
+      body: TypedValue,
       curried: scala.List[(Name, RTValue)],
       closingContext: CallStackFrame
   )
 
   case class FieldFunction(fieldName: Name) extends Function
 
-  case class LambdaFunction(body: RuntimeValue, pattern: Pattern[ValueAttribs], closingContext: CallStackFrame)
+  case class LambdaFunction(body: TypedValue, pattern: Pattern[UType], closingContext: CallStackFrame)
       extends Function
 
   case class DefinitionFunction(
-      body: RuntimeValue,
-      arguments: scala.List[(Name, ValueAttribs, Type[TypeAttribs])],
+      body: TypedValue,
+      arguments: scala.List[(Name, UType, Type[scala.Unit])],
```

Comment:
This one is a bit weird. The two types specified here are the same - I had to do some digging, but that is in fact the case. It comes from Morphir's `Definition` type, which takes both value attributes and a type for each input, because `Definition`s are typed whether or not `VA` represents that type, but `VA` could potentially hold some other information. I copied that format over into my `DefinitionFunction`, despite the fact that it never uses those types (it trusts the type checker did its job and just counts the number of applied vs. expected arguments).

Cleaning that up is a broader discussion, tho, so I would just change this to `UType, UType` and leave it at that for now.